### PR TITLE
hdf5 1.10.1 (import from homebrew/science)

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
We're going to want to tear out a lot of the platform-specific logic, some options and perhaps change the default options too.

CC @ilovezfs @imichka for thoughts and review.

```
install events in the last 365 days for homebrew/science/hdf5
==================================================================================================
  1 | homebrew/science/hdf5                                                    | 116,478 |  94.77%
  2 | homebrew/science/hdf5 --with-mpi                                         |   3,022 |   2.46%
  3 | homebrew/science/hdf5 --with-mpi --with-fortran                          |     623 |   0.51%
  4 | homebrew/science/hdf5 --with-fortran                                     |     508 |   0.41%
  5 | homebrew/science/hdf5 --c++11                                            |     426 |   0.35%
  6 | homebrew/science/hdf5 --with-mpi --c++11                                 |     231 |   0.19%
  7 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --with-fortran       |     220 |   0.18%
  8 | homebrew/science/hdf5 --with-fortran2003 --with-fortran                  |     118 |   0.10%
  9 | homebrew/science/hdf5 --with-mpi --c++11 --with-fortran                  |     103 |   0.08%
 10 | homebrew/science/hdf5 --c++11 --with-fortran                             |      76 |   0.06%
 11 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --c++11 --with-fortr |      58 |   0.05%
 12 | homebrew/science/hdf5 --with-fortran2003 --c++11 --with-fortran          |      56 |   0.05%
 13 | homebrew/science/hdf5 --with-threadsafe                                  |      55 |   0.04%
 14 | homebrew/science/hdf5 --c 11                                             |      42 |   0.03%
 15 | homebrew/science/hdf5 --with-test --with-mpi --with-fortran              |      39 |   0.03%
 16 | homebrew/science/hdf5 --universal                                        |      38 |   0.03%
 17 | homebrew/science/hdf5 --with-mpi --without-cxx                           |      33 |   0.03%
 18 | homebrew/science/hdf5 --with-test --with-mpi --c++11 --with-fortran      |      32 |   0.03%
 19 | homebrew/science/hdf5 --with-mpi --c 11                                  |      29 |   0.02%
 20 | homebrew/science/hdf5 --with-test --without-cxx                          |      29 |   0.02%
 21 | homebrew/science/hdf5 --with-test                                        |      25 |   0.02%
 22 | homebrew/science/hdf5 --with-test --with-fortran                         |      25 |   0.02%
 23 | homebrew/science/hdf5 --with-test --with-mpi                             |      24 |   0.02%
 24 | homebrew/science/hdf5 --with-threadsafe --with-unsupported               |      24 |   0.02%
 25 | homebrew/science/hdf5 --with-test --with-fortran2003 --with-mpi --c++11  |      22 |   0.02%
 26 | homebrew/science/hdf5 --with-threadsafe --with-fortran2003 --with-mpi -- |      18 |   0.01%
 27 | homebrew/science/hdf5 --with-mpi --with-unsupported --c++11 --with-fortr |      17 |   0.01%
 28 | homebrew/science/hdf5 --with-threadsafe --with-mpi                       |      17 |   0.01%
 29 | homebrew/science/hdf5 --with-threadsafe --with-fortran2003 --with-fortra |      16 |   0.01%
 30 | homebrew/science/hdf5 --with-threadsafe --with-unsupported --c++11       |      16 |   0.01%
 31 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --with-unsupported - |      15 |   0.01%
 32 | homebrew/science/hdf5 --with-fortran2003                                 |      13 |   0.01%
 33 | homebrew/science/hdf5 --with-mpi --with-unsupported --c++11              |      13 |   0.01%
 34 | homebrew/science/hdf5 --universal --c++11                                |      11 |   0.01%
 35 | homebrew/science/hdf5 --with-mpi --with-unsupported                      |      11 |   0.01%
 36 | homebrew/science/hdf5 --with-threadsafe --c++11                          |      11 |   0.01%
 37 | homebrew/science/hdf5 --with-threadsafe --with-mpi --with-unsupported -- |      11 |   0.01%
 38 | homebrew/science/hdf5 --with-check --with-fortran2003 --with-mpi --c++11 |      10 |   0.01%
 39 | homebrew/science/hdf5 --with-test --with-fortran2003 --with-mpi --with-f |      10 |   0.01%
 40 | homebrew/science/hdf5 --with-threadsafe --with-mpi --with-unsupported -- |      10 |   0.01%
 41 | homebrew/science/hdf5 --with-test --c++11 --with-fortran                 |       9 |   0.01%
 42 | homebrew/science/hdf5 --with-test --with-mpi --without-cxx               |       9 |   0.01%
 43 | homebrew/science/hdf5 --with-fortran2003 --c++11                         |       8 |   0.01%
 44 | homebrew/science/hdf5 --with-test --with-mpi --c++11                     |       8 |   0.01%
 45 | homebrew/science/hdf5 --with-threadsafe --with-fortran                   |       8 |   0.01%
 46 | homebrew/science/hdf5 --with-check --with-fortran2003 --with-mpi --c 11  |       7 |   0.01%
 47 | homebrew/science/hdf5 --with-fortran2003 --with-mpi                      |       7 |   0.01%
 48 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --without-cxx --with |       7 |   0.01%
 49 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi --with-un |       7 |   0.01%
 50 | homebrew/science/hdf5 --with-threadsafe --with-mpi --with-fortran        |       7 |   0.01%
 51 | homebrew/science/hdf5 --with-threadsafe --with-mpi --without-cxx --witho |       7 |   0.01%
 52 | homebrew/science/hdf5 --with-unsupported                                 |       7 |   0.01%
 53 | homebrew/science/hdf5 --without-cxx                                      |       7 |   0.01%
 54 | homebrew/science/hdf5 --with-check --with-fortran2003 --with-mpi --with- |       6 |   0.00%
 55 | homebrew/science/hdf5 --with-test --with-mpi --with-unsupported --with-f |       6 |   0.00%
 56 | homebrew/science/hdf5 --with-threadsafe --without-cxx                    |       6 |   0.00%
 57 | homebrew/science/hdf5 --with-threadsafe --without-cxx --without-hl       |       6 |   0.00%
 58 | homebrew/science/hdf5 --universal --with-fortran2003 --with-mpi --c++11  |       5 |   0.00%
 59 | homebrew/science/hdf5 --universal --with-mpi                             |       5 |   0.00%
 60 | homebrew/science/hdf5 --with-check                                       |       5 |   0.00%
 61 | homebrew/science/hdf5 --with-check --with-mpi --with-fortran             |       5 |   0.00%
 62 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --c 11 --with-fortra |       5 |   0.00%
 63 | homebrew/science/hdf5 --with-test --c++11                                |       5 |   0.00%
 64 | homebrew/science/hdf5 --with-test --with-mpi --with-unsupported --c++11  |       5 |   0.00%
 65 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi --c++11   |       5 |   0.00%
 66 | homebrew/science/hdf5 --with-threadsafe --with-mpi --c++11 --with-fortra |       5 |   0.00%
 67 | homebrew/science/hdf5 --with-threadsafe --without-cxx --with-unsupported |       5 |   0.00%
 68 | homebrew/science/hdf5 --without-cxx --c++11                              |       5 |   0.00%
 69 | homebrew/science/hdf5 --universal --with-test --with-threadsafe --with-f |       4 |   0.00%
 70 | homebrew/science/hdf5 --universal --with-test --with-threadsafe --with-u |       4 |   0.00%
 71 | homebrew/science/hdf5 --universal --with-threadsafe --with-mpi --c++11 - |       4 |   0.00%
 72 | homebrew/science/hdf5 --with-check --with-mpi --c 11 --with-fortran      |       4 |   0.00%
 73 | homebrew/science/hdf5 --with-fortran2003 --with-unsupported --c++11 --wi |       4 |   0.00%
 74 | homebrew/science/hdf5 --with-mpi --c 11 --with-fortran                   |       4 |   0.00%
 75 | homebrew/science/hdf5 --with-mpi --without-cxx --with-fortran            |       4 |   0.00%
 76 | homebrew/science/hdf5 --with-test --with-fortran2003 --c++11 --with-fort |       4 |   0.00%
 77 | homebrew/science/hdf5 --with-threadsafe --with-mpi --with-unsupported    |       4 |   0.00%
 78 | homebrew/science/hdf5 --with-threadsafe --with-unsupported --c 11        |       4 |   0.00%
 79 | homebrew/science/hdf5 --with-threadsafe --without-hl                     |       4 |   0.00%
 80 | homebrew/science/hdf5 --universal --with-mpi --with-fortran              |       3 |   0.00%
 81 | homebrew/science/hdf5 --universal --with-test --with-threadsafe --with-f |       3 |   0.00%
 82 | homebrew/science/hdf5 --with-check --c++11 --with-fortran                |       3 |   0.00%
 83 | homebrew/science/hdf5 --with-check --with-fortran2003 --c 11 --with-fort |       3 |   0.00%
 84 | homebrew/science/hdf5 --with-check --with-mpi                            |       3 |   0.00%
 85 | homebrew/science/hdf5 --with-check --with-threadsafe                     |       3 |   0.00%
 86 | homebrew/science/hdf5 --with-check --with-threadsafe --with-fortran2003  |       3 |   0.00%
 87 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --without-cxx --c++1 |       3 |   0.00%
 88 | homebrew/science/hdf5 --with-test --with-fortran2003 --with-fortran      |       3 |   0.00%
 89 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi --with-fo |       3 |   0.00%
 90 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi --with-un |       3 |   0.00%
 91 | homebrew/science/hdf5 --with-threadsafe --c 11                           |       3 |   0.00%
 92 | homebrew/science/hdf5 --with-threadsafe --with-fortran2003 --c++11 --wit |       3 |   0.00%
 93 | homebrew/science/hdf5 --with-threadsafe --without-cxx --without-hl --c++ |       3 |   0.00%
 94 | homebrew/science/hdf5 --with-unsupported --c++11                         |       3 |   0.00%
 95 | homebrew/science/hdf5 --c 11 --with-fortran                              |       2 |   0.00%
 96 | homebrew/science/hdf5 --universal --c 11 --with-fortran                  |       2 |   0.00%
 97 | homebrew/science/hdf5 --universal --c++11 --with-fortran                 |       2 |   0.00%
 98 | homebrew/science/hdf5 --universal --with-fortran                         |       2 |   0.00%
 99 | homebrew/science/hdf5 --universal --with-fortran2003 --c++11             |       2 |   0.00%
100 | homebrew/science/hdf5 --universal --with-fortran2003 --with-mpi --with-f |       2 |   0.00%
101 | homebrew/science/hdf5 --universal --with-test                            |       2 |   0.00%
102 | homebrew/science/hdf5 --universal --with-test --with-fortran             |       2 |   0.00%
103 | homebrew/science/hdf5 --universal --with-test --with-threadsafe --with-f |       2 |   0.00%
104 | homebrew/science/hdf5 --universal --with-threadsafe --with-fortran2003 - |       2 |   0.00%
105 | homebrew/science/hdf5 --with-check --c 11                                |       2 |   0.00%
106 | homebrew/science/hdf5 --with-check --c 11 --with-fortran                 |       2 |   0.00%
107 | homebrew/science/hdf5 --with-check --with-mpi --c++11                    |       2 |   0.00%
108 | homebrew/science/hdf5 --with-check --with-threadsafe --with-fortran2003  |       2 |   0.00%
109 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --with-unsupported - |       2 |   0.00%
110 | homebrew/science/hdf5 --with-mpi --without-hl --c++11 --with-fortran     |       2 |   0.00%
111 | homebrew/science/hdf5 --with-mpi --without-hl --with-fortran             |       2 |   0.00%
112 | homebrew/science/hdf5 --with-test --with-fortran2003 --with-mpi --c++11  |       2 |   0.00%
113 | homebrew/science/hdf5 --with-test --with-mpi --with-unsupported --c++11  |       2 |   0.00%
114 | homebrew/science/hdf5 --with-test --with-mpi --without-cxx --with-fortra |       2 |   0.00%
115 | homebrew/science/hdf5 --with-test --with-mpi --without-hl --with-fortran |       2 |   0.00%
116 | homebrew/science/hdf5 --with-test --with-threadsafe --with-fortran2003 - |       2 |   0.00%
117 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi --without |       2 |   0.00%
118 | homebrew/science/hdf5 --with-threadsafe --with-fortran2003 --with-mpi -- |       2 |   0.00%
119 | homebrew/science/hdf5 --with-threadsafe --with-mpi --c 11                |       2 |   0.00%
120 | homebrew/science/hdf5 --with-threadsafe --with-mpi --c++11               |       2 |   0.00%
121 | homebrew/science/hdf5 --with-threadsafe --with-unsupported --with-fortra |       2 |   0.00%
122 | homebrew/science/hdf5 --enable-fortran                                   |       1 |   0.00%
123 | homebrew/science/hdf5 --universal --c 11                                 |       1 |   0.00%
124 | homebrew/science/hdf5 --universal --with-check                           |       1 |   0.00%
125 | homebrew/science/hdf5 --universal --with-check --with-fortran2003 --c 11 |       1 |   0.00%
126 | homebrew/science/hdf5 --universal --with-check --with-fortran2003 --with |       1 |   0.00%
127 | homebrew/science/hdf5 --universal --with-check --with-mpi --with-fortran |       1 |   0.00%
128 | homebrew/science/hdf5 --universal --with-check --with-threadsafe --with- |       1 |   0.00%
129 | homebrew/science/hdf5 --universal --with-check --with-threadsafe --with- |       1 |   0.00%
130 | homebrew/science/hdf5 --universal --with-fortran2003 --c++11 --with-fort |       1 |   0.00%
131 | homebrew/science/hdf5 --universal --with-mpi --c 11 --with-fortran       |       1 |   0.00%
132 | homebrew/science/hdf5 --universal --with-mpi --c++11 --with-fortran      |       1 |   0.00%
133 | homebrew/science/hdf5 --universal --with-test --with-mpi --c++11         |       1 |   0.00%
134 | homebrew/science/hdf5 --universal --with-test --with-threadsafe --with-m |       1 |   0.00%
135 | homebrew/science/hdf5 --universal --with-threadsafe --with-unsupported - |       1 |   0.00%
136 | homebrew/science/hdf5 --universal --without-cxx --with-fortran           |       1 |   0.00%
137 | homebrew/science/hdf5 --with-check --with-fortran                        |       1 |   0.00%
138 | homebrew/science/hdf5 --with-check --with-fortran2003 --with-fortran     |       1 |   0.00%
139 | homebrew/science/hdf5 --with-check --with-mpi --c++11 --with-fortran     |       1 |   0.00%
140 | homebrew/science/hdf5 --with-check --with-threadsafe --with-fortran2003  |       1 |   0.00%
141 | homebrew/science/hdf5 --with-check --with-threadsafe --with-mpi --c 11 - |       1 |   0.00%
142 | homebrew/science/hdf5 --with-check --with-threadsafe --with-mpi --with-f |       1 |   0.00%
143 | homebrew/science/hdf5 --with-fortran --with-mpi                          |       1 |   0.00%
144 | homebrew/science/hdf5 --with-fortran2003 --c 11 --with-fortran           |       1 |   0.00%
145 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --without-cxx        |       1 |   0.00%
146 | homebrew/science/hdf5 --with-fortran2003 --with-mpi --without-mpi --with |       1 |   0.00%
147 | homebrew/science/hdf5 --with-mpi --with-unsupported --with-fortran       |       1 |   0.00%
148 | homebrew/science/hdf5 --with-netcdf                                      |       1 |   0.00%
149 | homebrew/science/hdf5 --with-test --with-mpi --with-unsupported          |       1 |   0.00%
150 | homebrew/science/hdf5 --with-test --with-threadsafe                      |       1 |   0.00%
151 | homebrew/science/hdf5 --with-test --with-threadsafe --with-fortran       |       1 |   0.00%
152 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi           |       1 |   0.00%
153 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi --c++11 - |       1 |   0.00%
154 | homebrew/science/hdf5 --with-test --with-threadsafe --with-mpi --without |       1 |   0.00%
155 | homebrew/science/hdf5 --with-test --with-threadsafe --with-unsupported - |       1 |   0.00%
156 | homebrew/science/hdf5 --with-test --with-unsupported --c++11 --with-fort |       1 |   0.00%
157 | homebrew/science/hdf5 --with-threadsafe --c++11 --with-fortran           |       1 |   0.00%
158 | homebrew/science/hdf5 --with-threadsafe --with-fortran2003 --with-mpi -- |       1 |   0.00%
159 | homebrew/science/hdf5 --with-threadsafe --with-mpi --with-unsupported -- |       1 |   0.00%
160 | homebrew/science/hdf5 --with-threadsafe --with-mpi --without-cxx --witho |       1 |   0.00%
161 | homebrew/science/hdf5 --with-threadsafe --with-mpi --without-hl          |       1 |   0.00%
162 | homebrew/science/hdf5 --with-threadsafe --with-mpi --without-hl --c++11  |       1 |   0.00%
163 | homebrew/science/hdf5 --with-threadsafe --with-mpi --without-hl --c++11  |       1 |   0.00%
164 | homebrew/science/hdf5 --with-threadsafe --without-cxx --c 11             |       1 |   0.00%
165 | homebrew/science/hdf5 --with-threadsafe --without-hl --c++11             |       1 |   0.00%
166 | homebrew/science/hdf5 --with-threadsafe --without-hl --without-cxx       |       1 |   0.00%
167 | homebrew/science/hdf5 --with-unsupported --c 11                          |       1 |   0.00%
168 | homebrew/science/hdf5 --with-unsupported --c++11 --with-fortran          |       1 |   0.00%
==================================================================================================
Total                                                                          | 122,902 |    100%
==================================================================================================
```

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Worth noting: things in homebrew/science depend on `with-cx11` and `with-mpi` options so we may need to consider making them the defaults.

**Comment by iMichka** [comment]
One note about `with-mpi`: it breaks some formulae that do not need the mpi support. This is why it was never added as a default. I can make a list but it may take me some time. For 'with-cx11' I do not know, but I guess that is waaaay to recent c++ for most of the science formulae. We would need to test that too.

The same holds for `with-threadsafe` and `with-fortran2003`. We need to be careful because some options are exclusive, and I think these also break different builds.

We have an issue open for all these options: https://github.com/Homebrew/homebrew-science/issues/5470, but didn't have time to look into it. I can spare some free time on Saturday and Sunday to look into the various hdf5 options.

**Comment by ilovezfs** [comment]
I think `mpi` and `fortran` default enabled is the way to go for science formulae with either or both of those options.

**Comment by iMichka** [comment]
@MikeMcQuaid @ilovezfs Once hdf5 is merged, I'll migrate one of the vtk/opencv/opencv3 formulae

**Comment by iMichka** [comment]
Do not bother about with-fortran2003, that option is already gone, as it is unsupported on newer hdf5 versions: https://github.com/Homebrew/homebrew-science/pull/5294

**Comment by iMichka** [comment]
@MikeMcQuaid hdf5 needs to be removed from tap_migrations.json.

> --c++11
	
We should use this by default.

> --with-fortran

That should be default too. Didn't test if it breaks stuff though.

> --with-mpi

 I am going through some builds to see if the --with-mpi flag is doing any harm but it looks pretty good I think. I thought it would break stuff, but in fact it seems not. If there are 1 or 2 formulae broken due to this along the road, I'm ready to work on an upstream patch.

> --with-test

Let's drop this. (same as for the `R` formula)

--with-threadsafe (Trade performance for C API thread-safety (requires --without-cxx and --without-hl)

Not needed by any formulae in science. Not compatible with mpi. It's quite a niche thing, but still nice to have.

> --with-unsupported Allow unsupported combinations of configure options

Meh?

--without-cxx Disable the C++ interface

Probably just exists due to `--with-threadsafe`. If `--with-threadsafe` is dropped, this can go too.

--without-hl (Compile without high-level library)

As for `--without-cxx`, is only there for `--with-threadsafe`.

So basically we can have `--with-mpi` by default. For `--with-threadsafe`, either it stays as an option, either it is dropped, either we create a `hdf5-threadsafe` formula.

**Comment by iMichka** [comment]
Talked too fast. I hit the 2 first failures:

xmi-msim:
```
checking for h5cc-64... no
checking for h5cc... no
configure: error: An installation of HDF5 is required, including the headers
```

anvio:
```
/usr/local/include/H5public.h:59:13: fatal error: 'mpi.h' file not found
    #   include <mpi.h>
                ^
    1 error generated.
```

Can probably be fixed though.

**Comment by tschoonj** [comment]
I am not in favour of making --with-mpi a default option as it drags in a big dependency that will most likely not be used by a lot of people, especially on macOS (Linux is a different matter...)

Also, is parallel HDF5 guaranteed to work properly with code that is meant for serial HDF5? Has anyone done testing on this?

**Comment by ilovezfs** [comment]
Here are analytics on the formulae that need hdf5 with mpi for number of installs in the last 30 days:
```
petsc   177
slepc    96
trilinos 53
abinit   41
dealii   35
getdp     3
```

**Comment by iMichka** [comment]
> Also, is parallel HDF5 guaranteed to work properly with code that is meant for serial HDF5? Has anyone done testing on this?

Do you mean MPI or thread safe?

Let me try to elaborate (at least from what I know, and with the help of [stackoverflow](https://stackoverflow.com/questions/34906652/does-hdf5-support-concurrent-reads-or-writes-to-different-files))

You can build:
* with neither thread-safety nor MPI support (what we do; but what also conda does for example)
* with MPI support but no thread safety
* with thread safety but no MPI support

Disadvantages of the `thread safe` option: **it is slower**, and there is no windows support for it. This is why that option is really not commonly used. We have no other formula needing that kind of build.

MPI: needs some more code to be written to be supported. Should IMHO not break anything, but as shown above, I already found 2 broken formulae (I tested around 15 out of ~50). This is because we will need to pass the right compilation flags, so it will be some work to make sure all the formulae are really working with MPI. I can also not guarantee that there will be no other compilation problems down the road, I just made some preliminary tests.

Besides that, I have no clue on the real effect on runtime performance if we use one of both options. This is often the responsibility of upstream to carefully benchmark their software if it is time critical.

**Comment by MikeMcQuaid** [comment]
@iMichka @ilovezfs and anyone else with an `i` prefix: how's this look to you now?

**Comment by ilovezfs** [comment]
The Linux bottle in the bottle block is making me feel not well.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: I'm skeptical this line is actually needed for it to find zlib.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: This should move down, according to build before non-optional, and non-optional before optional.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: This should move down, according to build before non-optional, and non-optional before optional.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: This should move down, according to build before non-optional, and non-optional before optional.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: This is only to appease that awful audit, I think.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: Ditto. That audit is from the :smiling_imp:

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: yeah it finds it without this

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: This line isn't needed.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: The default option text for mpi is `Build with mpi support`. Do we think that actually needs further elaboration?

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: should also add `--disable-silent-rules`

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: don't need the `=yes` part

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: ditto

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: formatting could use some improvement here

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: Looks like the `ENV.cxx11 if build.cxx11?` is missing

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: would move this below `--disable-silent-rules`

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: shouldn't be spaces between deps

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: I'd move this below the ENV[]s with a blank line above it.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: Actually I think both `--enable-static` and `--enable-shared` are already the defaults so neither should be needed.

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: :mask:

**Comment by iMichka** [comment]
FYI, I won't be able to review this, I'm on holidays for two weeks. And I won't be able to babysit the broken formulae in science, if there are some. I do not know if you can hold this off for 2 weeks, until I come back? :)

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: szip should be above fortran and below libtool

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: can be deleted

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: can be deleted

File: Formula/hdf5.rb
Comment (ilovezfs) [inline Formula/hdf5.rb]: Looks like the ENV.cxx11 if build.cxx11? is missing

**Review by ilovezfs — APPROVED** [review]

**Comment by Y--** [comment]
Hi guys, not sure where is the appropriate place for me to raise the issue, please let me know if here isn't, but following this migration the C++ headers of the HDF5 lib aren't installed anymore.

As I am not an expert with brew I am not sure what the correct fix is. This version was working fine (beside the outdated hash): https://github.com/Homebrew/homebrew-science/find/af0844cb6ed0ac6b5f07a6a3d30a507de9d8a323

Please let me know what you suggest we should do here.

Thanks!

**Comment by ilovezfs** [comment]
@Y-- https://github.com/Homebrew/homebrew-core/pull/15771

**Comment by Y--** [comment]
Awesome, thanks!

**Comment by Y--** [comment]
It works indeed, thanks a lot for the quick fix!

**Pull request closed without merging**
